broker/submissions/export_to_spreadsheet_service.py

```diff
@@ -17,3 +24,44 @@ def export(self, submission_uuid: str):
         flattened_json = self.downloader.convert_json(entity_list)
         workbook = self.downloader.create_workbook(flattened_json)
         return workbook
+
+    def export_and_save(self, submission_uuid: str, storage_dir: str):
+        self.logger.info(f'Exporting submission {submission_uuid}')
+        try:
+            submission = self.ingest_api.get_submission_by_uuid(submission_uuid)
+            submission_url = submission['_links']['self']['href']
+        except Exception as e:
+            self.logger.error(e)
+            raise Exception(f'An error occurred in retrieving the submission with uuid {submission_uuid}: {str(e)}')
+
+        create_date = datetime.now(timezone.utc)
+
+        self._patch(submission_url, create_date)
+
+        directory = f'{storage_dir}/{submission_uuid}'
+        os.makedirs(f'{directory}/downloads/', exist_ok=True)
+
+        timestamp = create_date.strftime("%Y%m%d-%H%M%S")
+        filename = f'{submission_uuid}_{timestamp}.xlsx'
+        filepath = f'{directory}/downloads/{filename}'
+
+        workbook = self.export(submission_uuid)
+        workbook.save(filepath)
+
+        finished_date = datetime.now(timezone.utc)
+        self._patch(submission_url, create_date, finished_date)
+
+        self.logger.info(f'Done exporting spreadsheet for submission {submission_uuid}!')
+
+    def _patch(self, submission_url, create_date, finished_date=None):
+        patch = {
+            'lastSpreadsheetGenerationJob': {
+                'finishedDate': finished_date.isoformat().replace("+00:00", "Z") if finished_date else None,
```

Comment:
Why use `.isoformat().replace("+00:00", "Z")` twice and not use a function?

Reply:
I added a function to the date util in ingest-client and used it here.
Please review: https://github.com/ebi-ait/ingest-client/pull/41/files